client.go

```diff
@@ -42,7 +44,7 @@ type ClientConfig struct {
 	RTO            time.Duration
 	Conn           net.PacketConn // Listening socket (net.PacketConn)
 	LoggerFactory  logging.LoggerFactory
-	Net            *vnet.Net
+	Net            transport.Net
```

Comment:
Hi @enobufs,

The old `vnet.Net` does not implement the full `transport.Net` interface as we added and changed more functions to `transport.Net` which makes it more aligned with the standard `net` module.

Here is a diff of the two types:

```diff
--- old vnet.Net
+++ new transport.Net
-type Net interface {
+type Net struct {
 	CreateDialer(dialer *net.Dialer) Dialer
 	Dial(network, address string) (net.Conn, error)
-	DialTCP(network string, laddr, raddr *net.TCPAddr) (TCPConn, error)
-	DialUDP(network string, laddr, raddr *net.UDPAddr) (UDPConn, error)
-	InterfaceByIndex(index int) (*Interface, error)
+	DialUDP(network string, laddr, raddr *net.UDPAddr) (UDPPacketConn, error)
 	InterfaceByName(name string) (*Interface, error)
 	Interfaces() ([]*Interface, error)
+	IsVirtual() bool
 	ListenPacket(network string, address string) (net.PacketConn, error)
-	ListenTCP(network string, laddr *net.TCPAddr) (TCPListener, error)
-	ListenUDP(network string, locAddr *net.UDPAddr) (UDPConn, error)
-	ResolveIPAddr(network, address string) (*net.IPAddr, error)
-	ResolveTCPAddr(network, address string) (*net.TCPAddr, error)
+	ListenUDP(network string, locAddr *net.UDPAddr) (UDPPacketConn, error)
 	ResolveUDPAddr(network, address string) (*net.UDPAddr, error)
 }
```

Reply:
Here is the output of [go-apidiff](https://github.com/joelanford/go-apidiff):

```bash
stv0g@ubuntu:~/workspace/pion/transport$ go-apidiff v0.13.1 v0.14.0

github.com/pion/transport/vnet
  Incompatible changes:
  - (*Net).DialUDP: changed from func(string, *net.UDPAddr, *net.UDPAddr) (UDPPacketConn, error) to func(string, *net.UDPAddr, *net.UDPAddr) (github.com/pion/transport.UDPConn, error)
  - (*Net).IsVirtual: removed
  - (*Net).ListenUDP: changed from func(string, *net.UDPAddr) (UDPPacketConn, error) to func(string, *net.UDPAddr) (github.com/pion/transport.UDPConn, error)
  - Dialer: removed
  - Interface: removed
  - InterfaceBase: removed
  - NATType.Hairpining: removed
  - NewInterface: removed
  - NewNet: changed from func(*NetConfig) *Net to func(*NetConfig) (*Net, error)
  - UDPPacketConn: removed
```

Reply:
@Sean-Der retagged pion/transport with a v2 version tag to help users struggling with the incompatibilities.

Hence, I think we don't need to be worried to much for pion/turn.

Since, pion/transport is exposed by pion/turn through the `ClientConfig` struct, we will probably also need to bump the major version of pion/turn